src/server/jattr_get_set.c

```diff
@@ -174,11 +181,45 @@ char *
 get_jattr_str(const job *pjob, int attr_idx)
 {
 	if (pjob != NULL)
-		return pjob->ji_wattr[attr_idx].at_val.at_str;
+		return get_attr_str(get_jattr(pjob, attr_idx));
+
+	return NULL;
+}
+
+/**
+ * @brief	Getter function for job attribute of type string of array
+ *
+ * @param[in]	pjob - pointer to the job
+ * @param[in]	attr_idx - index of the attribute to return
+ *
+ * @return	struct array_strings *
+ * @retval	value of the attribute
+ * @retval	NULL if pjob is NULL
+ */
+struct array_strings *
+get_jattr_arst(const job *pjob, int attr_idx)
+{
+	if (pjob != NULL)
+		return get_attr_arst(get_jattr(pjob, attr_idx));
 
 	return NULL;
 }
 
+/**
+ * @brief	Getter for job attribute's list value
+ *
+ * @param[in]	pjob - pointer to the job
+ * @param[in]	attr_idx - index of the attribute to return
+ *
+ * @return	pbs_list_head
+ * @retval	value of attribute
+ */
+pbs_list_head
+get_jattr_list(const job *pjob, int attr_idx)
+{
+	return get_attr_list(get_jattr(pjob, attr_idx));
```

Comment:
do NULL check on pjob

Reply:
`get_attr_list` will return dummy if `pjob` is NULL.